Sources/Kitura/RouterRequest.swift

```diff
@@ -130,7 +128,30 @@ public class RouterRequest {
     public internal(set) var parameters: [String:String] = [:]
 
     /// List of query parameters.
-    public var queryParameters: [String:String] { return parsedURL.queryParameters }
+    public lazy var queryParameters: [String:String] = { [unowned self] in
+        var qParams: [String:String] = [:]
+        /*
+        if let queryItems = self.urlComponents.queryItems {
+            for item in queryItems {
+                qParams[item.name] = item.value
+            }
+        }
+        */
+        // urlComponents.queryItems above crashes on linux as of swift 3.0.1
```

Comment:
@ianpartridge  I believe it is related to https://bugs.swift.org/browse/SR-2462
And I think apple/swift-corelibs-foundation#719 contains a workaround for that bug addressing this specific property.